dask_cuda/proxify_host_file.py

```diff
@@ -570,31 +595,23 @@ def serialize_proxy_to_disk_inplace(cls, proxy: ProxyObject):
         proxy : ProxyObject
             Proxy object to serialize using the "disk" serialize.
         """
-        manager = proxy._obj_pxy_get_manager()
-        with manager.lock:
-            if not proxy._obj_pxy_is_serialized():
-                proxy._obj_pxy_serialize(serializers=("disk",))
-            else:
-                header, frames = proxy._obj_pxy["obj"]
-                if header["serializer"] in ("dask", "pickle"):
-                    path = cls.gen_file_path()
-                    with open(path, "wb") as f:
-                        f.write(pack_frames(frames))
-                    proxy._obj_pxy["obj"] = (
-                        {
-                            "serializer": "disk",
-                            "path": path,
-                            "shared-filesystem": cls._spill_shared_filesystem,
-                            "disk-sub-header": header,
-                        },
-                        [],
-                    )
-                    proxy._obj_pxy["serializer"] = "disk"
-                    manager.move(
-                        proxy,
-                        from_serializer=header["serializer"],
-                        to_serializer="disk",
-                    )
-                elif header["serializer"] != "disk":
-                    proxy._obj_pxy_deserialize()
-                    proxy._obj_pxy_serialize(serializers=("disk",))
+        pxy = proxy._pxy_get(copy=True)
+        if pxy.is_serialized():
+            header, frames = pxy.obj
+            if header["serializer"] in ("dask", "pickle"):
+                path = cls.gen_file_path()
+                with open(path, "wb") as f:
+                    f.write(pack_frames(frames))
+                pxy.obj = (
+                    {
+                        "serializer": "disk",
+                        "path": path,
+                        "shared-filesystem": cls._spill_shared_filesystem,
+                        "disk-sub-header": header,
+                    },
+                    [],
+                )
+                pxy.serializer = "disk"
+                proxy._pxy_set(pxy)
+                return
+        proxy._pxy_serialize(serializers=("disk",), proxy_detail=pxy)
```

Comment:
It is not exactly the same. Notice, we want to call `proxy._pxy_serialize()` if
```python
pxy.is_serialized() is False
```
OR
```python
header["serializer"] not in ("dask", "pickle")
```
And we have to check that
```python 
pxy.is_serialized() is True
```
before _unpacking_ `header, frames = pxy.obj` otherwise `pxy.obj` could be a regular object.

It is because I am _overloading_ the use of `pxy.obj` to either refer to the serialized object and the non-serialized object. I agree, this is a bit messy so maybe I should do a follow up PR that splits `pxy.obj`  into two different attributes. This would also make type hinting easier.

Reply:
Yeah, I think a follow-up PR would be good. The way it's written now it's totally not obvious that `header, frames = pxy.obj` is a necessary step before calling `proxy._pxy_serialize(serializers=("disk",), proxy_detail=pxy)`.